Content.Server/Connection/ConnectionManager.cs

```diff
@@ -168,20 +174,36 @@ private async Task NetMgrOnConnecting(NetConnectingArgs e)
                 return (ConnectionDenyReason.Ban, message, bans);
             }
 
-            if (_cfg.GetCVar(CCVars.WhitelistEnabled))
+            // Checks for whitelist IF it's enabled AND the user isn't an admin. Admins are always allowed.
+            if (_cfg.GetCVar(CCVars.WhitelistEnabled) && adminData is null)
             {
-                var min = _cfg.GetCVar(CCVars.WhitelistMinPlayers);
-                var max = _cfg.GetCVar(CCVars.WhitelistMaxPlayers);
-                var playerCountValid = _plyMgr.PlayerCount >= min && _plyMgr.PlayerCount < max;
+                var whitelists = _prototypeManager.EnumeratePrototypes<PlayerConnectionWhitelistPrototype>();
+                var whitelistStringList = _cfg.GetCVar(CCVars.WhitelistPrototype);
 
-                if (playerCountValid && await _db.GetWhitelistStatusAsync(userId) == false
-                                     && adminData is null)
+                // The whitelistStringList is a comma separated list of whitelists to check. I loop through every prototype in the list and check if the user is whitelisted.
+                foreach (var whitelistString in whitelistStringList.Split(','))
```

Comment:
Do all this string splitting and figuring out the prototypes ahead of time with an OnValueChanged.